easysnmp/interface.c

```diff
@@ -1142,12 +1142,12 @@ static int __send_sync_pdu(netsnmp_session *ss, netsnmp_pdu *pdu,
                            bitarray *invalid_oids)
 {
     int status = 0;
-    long command = pdu->command;
```

Comment:
Revert this change. It doesn't align with the definition of `netsnmp_pdu`: http://net-snmp.sourceforge.net/dev/agent/structsnmp__pdu.html .

Reply:
I've seen **int** type for **command** field in source file. In orig code long type now. I don't quite understand. Need to go back to long again?

Reply:
*sigh* ok, this should be resolved in a set of follow up commits. My concern is that there are going to be compiler errors triggered by `-Woverflow`, etc, but the problems will need to be resolved in a single shot.

Reply:
> _sigh_ ok, this should be resolved in a set of follow up commits. My concern is that there are going to be compiler errors triggered by `-Woverflow`, etc, but the problems will need to be resolved in a single shot.

I'd rather agree. But I don't fully understand this part. I'm sorry, please, if you have time to explain this section to me i'll be grateful. Why do we need type "long" if in the source header "command" field have "int" type? Will there be an overflow when we copying from long to int in netsnmp_pdu structure?

I mean, if i have:
long command = pdu->command;

pdu->command have "int" type, but in "command" var I have "long" type.
If i will use two different type - it's seems like overflow can happen, instead of if I will use the same type for both - than there should be no overflow.

Is it right?
Or maybe you can advise me some source where i can read and learn about it and find my answers. Thanks.

Reply:
It variable uses later in "snmp_fix_pdu" function in second parameter. And second parameter accepts int too

Reply:
> Will there be an overflow when we copying from long to int in netsnmp_pdu structure?

There could be if we're not careful (monotonic counters are notorious for causing problems).

However, if the underlying type for some of the APIs is already int, then the issue will need to be addressed in the library as another series of commits to prevent overflow from happening when an int overflows, as (2^32) / 2-1 -> 2,147,483,647, whereas (2^64)/2 - 1 -> 9.223372036854776e+18.

Reply:
Thank you, i did